src/__integrations__/docker/payWorkflow.browser.integration.docker.test.ts

```diff
@@ -0,0 +1,248 @@
+    await page.focus(creditCardHolderFieldS);
+    await page.keyboard.type('Luigi XIV');
+
+    const creditCardPANFieldS = '#creditcardnumber';
+    await page.waitForSelector(creditCardPANFieldS);
+    await page.focus(creditCardPANFieldS);
+    await page.keyboard.type('4024007182788397');
+
+    const creditCardExpDateFieldS = '#creditcardexpirationdate';
+    await page.waitForSelector(creditCardExpDateFieldS);
+    await page.focus(creditCardExpDateFieldS);
+    await page.keyboard.type('01/25');
+
+    const creditCardSecurCodeFieldS = '#creditcardsecurcode';
+    await page.waitForSelector(creditCardSecurCodeFieldS);
+    await page.focus(creditCardSecurCodeFieldS);
+    await page.keyboard.type('666');
+
+    const privacyToggleS = '#creditcardform #privacyToggler';
+    await page.waitForSelector(privacyToggleS);
+    await page.click(privacyToggleS);
+
+    const buttonS = '#creditcardform > .windowcont__bottom > .container > .windowcont__bottom__wrap > .btn-primary';
+    await page.waitForSelector(buttonS);
+
+    const serverResponse = await Promise.all([
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /start-session/.test(response.request().url()),
+      ),
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /approve-terms/.test(response.request().url()),
+      ),
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /wallet/.test(response.request().url()),
+      ),
+      page.click(buttonS),
+      page.waitForNavigation(),
+    ]);
+
+    // Assert CORS is working
+
+    expect(serverResponse[0]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+    expect(serverResponse[1]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+    expect(serverResponse[2]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+
+    const sessionStorageState = await Promise.all([
+      page.evaluate(() => sessionStorage.getItem('wallet')),
+      page.evaluate(() => sessionStorage.getItem('sessionToken')),
+      page.evaluate(() => sessionStorage.getItem('approvalState')),
+    ]);
+
+    // Assert response payload, stored in Session Storage, is correct
+    expect(
+      fromNullable(sessionStorageState[0])
+        .map(myString => JSON.parse(myString))
+        .getOrElse({}),
+    ).toMatchObject({ creditCard: { holder: 'Luigi XIV' } });
+
+    expect(
+      fromNullable(sessionStorageState[1])
+        .map(myString => /[\d\w]{128}/.test(myString))
+        .getOrElse(false),
+    ).toBeTruthy();
+
+    expect(
+      fromNullable(sessionStorageState[2])
+        .map(myString => JSON.parse(myString))
+        .getOrElse({}),
+    ).toMatchObject({ acceptTerms: true });
+
+    await page.close();
+  });
+
+  it('should call start check payment, when app is loaded', async () => {
```

Comment:
This test should run before `should call start session, approve terms and add wallet when Continua is pressed on Credit Card input form`, or not ?

Reply:
@pasqualespica OK, I'll fix the test order